src/components/AssetsComponent.jsx

```diff
@@ -78,13 +77,7 @@ export default class AssetsComponent extends Component {
 
   retrieveAssets = (activePage, limit, status, filters = {}) => {
     if (checkIfCutoffExceeded(activePage, limit)) {
-      const url = constructUrl(activePage, limit, filters, status);
-
-      this.props.loading(true);
-      return fetchData(url).then((response) => {
-        this.props.loading(false);
-        this.setState({ assets: response.data.results });
-      });
+      this.props.getAssetsAction(activePage, limit, filters, status);

```

Comment:
@rotimi-babalola the logic you have changed have something to do with the new pagination implementation. The logic ensured that the application was not using too much memory of the user's device. Imagine having over 500 items in memory in more than 5 pages - this will cause performance issues. That's why we had to have a tradeoff between performance and memory, and the logic you deleted fixed the issue.

Please revert back to the previous changes.

If you look at the render function, the `assets` being displayed are obtained either from props or the `assets` state. Therefore, instead of checking for empty assets list in [the container](https://github.com/AndelaOSP/art-dashboard/blob/develop/src/_components/Assets/AssetsContainer.jsx#L62), `shouldFetchAssets` should be moved to the component in order to use the state variable. Therefore, we might have something like:
```jsx
checkIfShouldFetchAssets = () => {
  const { activePage, assetsList, shouldReload } = this.props;
  const pageKey = `page_${activePage}`;
  const activePageAssets = assetsList[pageKey] || this.state.assets;

  return shouldReload || isEmpty(activePageAssets);
}
```

Then in `componentDidMount`, have the following:
```jsx
const { selected } = this.props;
const shouldFetchAssets = this.checkIfShouldFetchAssets();
if (shouldFetchAssets) {
   this.retrieveAssets(activePage, this.state.limit, selected, status);
}
```
Note that I'm using `retrieveAssets` as it takes into account 2 use cases:
- Fetching directly from the API
- Use of redux action

Also, I have realized that similar bugs might arise in the component due to the way the logic is. Therefore, replace all instances of `this.loadAssets(activePage, data.value, null, status);`, except the one in `retrieveAssets` with `this.retrieveAssets(...)`.  Don't forget to pass the `filters` parameter. 

As for `this.loadAssets` in `retrieveAssets`, replace it with the action creator. I.e.
```jsx
 retrieveAssets = (activePage, limit, status, filters = {}) => {
    if (checkIfCutoffExceeded(activePage, limit)) {
      ...
    }

    return this.props.getAssetsAction(activePage, limit, filters, status);
  };
```

Don't forget to delete `loadAssets = (activePage, limit, filters, status) => {...}` afterwards.